### PR TITLE
Add Kubernetes TLS configuration documentation.

### PR DESCRIPTION
**Pull request by MirahImage**
Additional documentation in order to reduce user struggle configuring TLS for RabbitMQ clusters on Kubernetes.

### CODE REVIEW

File: site/kubernetes/operator/using-operator.md [hunk 1]
```diff
@@ -940,6 +941,88 @@ For more information about concepts mentioned above, see:
   </col>
 </table>
 
+## <a id='tls' class='anchor' href='#tls'>(Optional) Configure TLS</a>
+
+Transport Layer Security (TLS) is a protocol for encrypting network traffic. The cluster operator supports
+configuring a RabbitMQ cluster to support <a href="#one-way-tls">TLS</a> or <a href="#mutual-tls">mutual TLS (mTLS)</a> encrypted traffic between clients and the cluter, as well as supporting <a href="https://github.com/rabbitmq/cluster-operator/tree/main/docs/examples/mtls-inter-node">encrypting RabbitMQ inter-node traffic with mTLS</a>.
```
Comment (michaelklishin) [inline site/kubernetes/operator/using-operator.md:947]: Can we work in a link to our main TLS guide here, please?

File: site/kubernetes/operator/using-operator.md [hunk 2]
```diff
@@ -940,6 +941,88 @@ For more information about concepts mentioned above, see:
...
+Depending on the service type used (`spec.service.type`), further SAN attributes may be required.
+For example, if using service type `NodePort`, the SAN must include the external IP address of each Kubernetes node.
+
+To enable TLS, create a Kubernetes secret containing the pem encoded server certificate `server.pem` and private key `server-key.pem`
+
+<pre class='lang-bash'>
+kubectl create secret tls tls-secret --cert=server.pem --key=server-key.pem
+</pre>
+
+or use a tool such as <a href="https://cert-manager.io/">Cert Manger</a> to generate a tls secret.
```
Comment (michaelklishin) [inline site/kubernetes/operator/using-operator.md:974]: TLS secret

File: site/kubernetes/operator/using-operator.md [hunk 3]
```diff
@@ -940,6 +941,88 @@ For more information about concepts mentioned above, see:
...
+* `myrabbit-server-0.myrabbit-nodes.mynamespace.svc.cluster.local`
+* `myrabbit-server-1.myrabbit-nodes.mynamespace.svc.cluster.local`
+* `myrabbit-server-2.myrabbit-nodes.mynamespace.svc.cluster.local`
+* `myrabbit.mynamespace.svc.cluster.local`
+
+Note that the last SAN attribute is the client service DNS name.
+Depending on the service type used (`spec.service.type`), further SAN attributes may be required.
+For example, if using service type `NodePort`, the SAN must include the external IP address of each Kubernetes node.
+
+To enable TLS, create a Kubernetes secret containing the pem encoded server certificate `server.pem` and private key `server-key.pem`
```
Comment (michaelklishin) [inline site/kubernetes/operator/using-operator.md:968]: PEM-encoded

File: site/kubernetes/operator/using-operator.md [hunk 4]
```diff
@@ -940,6 +941,88 @@ For more information about concepts mentioned above, see:
...
+metadata:
+  name: mtls
+spec:
+  replicas: 1
+  tls:
+    secretName: tls-secret
+    caSecretName: ca-secret
+</pre>
+
+In order to enforce client verification, RabbitMQ must be configured to reject clients that do not present certificates. This can be done by enabling `ssl.options.fail_if_no_peer_cert` in the additional config:
```
Comment (michaelklishin) [inline site/kubernetes/operator/using-operator.md:1017]: `ssl.options` should be `ssl_options`. We should also link to [this TLS guide section](https://www.rabbitmq.com/ssl.html#peer-verification), most users have a fairly vague idea of what peer verification really is.

**Review by michaelklishin — CHANGES_REQUESTED** [review]
This is a good start but could benefit from more linking and also has a few typos. We can link to this section from the main TLS guide, e.g. "for enabling TLS for client connections on Kubernetes, see this section in the Kubernetes Operator guide"

**Comment by MirahImage** [comment]
I have made the requested changes and added a link from the TLS page to this documentation.

File: site/kubernetes/operator/using-operator.md (hunk 1 above)
Comment (michaelklishin) [reply site/kubernetes/operator/using-operator.md]: Sounds good

**Review by michaelklishin — APPROVED** [review]

**Pull request merged**